ui/src/views/Quickstart/index.jsx

```diff
@@ -207,32 +205,68 @@ export default class QuickStart extends Component {
     this.setState(initialState);
   };
 
+  getMatchCondition = events => {
+    let condition = '';
+    const eventsJoin = Array.from(events).join(' ');
+
+    if (eventsJoin.includes('pull_request')) {
+      condition = `${condition}(tasks_for == "github-pull-request" && event["action"] in [${[
+        ...events,
+      ].sort()}])`;
+    }
+
+    if (eventsJoin.includes('push')) {
+      if (condition.length > 0) {
+        condition = `${condition} || `;
+      }
+
+      condition = `${condition}(tasks_for == "github-push")`;
+    }
+
+    if (eventsJoin.includes('release')) {
+      if (condition.length > 0) {
+        condition = `${condition} || `;
+      }
+
+      condition = `${condition}(tasks_for == "github-relase")`;
+    }
+
+    return condition;
+  };
+
   renderEditor() {
+    const {
+      access,
+      commands,
+      events,
+      image,
+      taskName,
+      taskDescription,
+    } = this.state;
     const newYaml = safeDump({
       ...initialYaml,
-      access: this.state.access,
-      tasks: [
-        {
-          ...initialYaml.tasks[0],
-          ...{
-            metadata: {
-              ...initialYaml.tasks[0].metadata,
-              name: this.state.taskName,
-              description: this.state.taskDescription,
-            },
-            extra: {
-              github: {
-                events: [...this.state.events].sort(),
+      policy: {
+        pullRequests: access,
+      },
+      tasks: {
+        $match: {
+          [`${this.getMatchCondition(events)}`]: {
```

Comment:
Th function is being run every time a user types in the editor which is not ideal (editor becomes slow when typing). Given that we only require `events` to determine the condition, we should probably be able to run this method whenever `handleEventsSelection` is triggered. Perhaps something like:

```jsx

  handleEventsSelection = ({ target: { value } }) => {
    const events = new Set(this.state.events);
    const condition = getMatchesCondition(events);

    events.has(value) ? events.delete(value) : events.add(value);

    this.setState({
      events,
      condition,
      editorValue: null,
    });
  };
```

Reply:
Doing this, its making the initial value of condition to null. As render function is using `condition` from state and initially that value is null.

Reply:
Fixed it